ForestAdmin/devops
# feat(Click Up): Automatically update status of clickup cards (#4prrm9)

# What is Done ?
- The status will be set to `doing` when creating a pull request if that's not already the case
- The status will pass to `waiting for review` when the assignee is a reviewer
- The status will pass to `on code review` once a review has been submitted
- The status will pass to `to release` once approved
- The status will pass to `released` once push on master

# What does not work yet
- The update of subtasks
  - I'm waiting for a reply from Click Up Support

# What is required for this action to works
Simply the task id in the title of the pull request and in the title when merging too.

# How does it works with subtasks
A task with subtasks will never be attached to a PR, only the subtasks. So whenever a subtasks is updated the parent will correctly be updated depending on all the subtasks
And when a tasks pass on released state the subtasks will too

# How to test
- Create a fake  card on click up. (Let the status in `todo`)
- Create a fake pull request on top of this one and in the title put `(#${ID_OF_MY_FAKE_TASK})`
- The status of the card should be in `doing` now
- Ask a review from someone on your fake PR
- The task should pass to `waiting for review`
- Ask the reviewer to do one comment and reject for example
- The task should be `in code review`
- Then ask the reviewer to approve the PR
- The task should be on `to release`
